src/headless/HeadlessUtils.h

```diff
@@ -0,0 +1,25 @@
+/*
+** HeadlessUtils provides a collection of utility functions to manage and output
+** data and construct and manipulate surge
+*/
+#pragma once
+
+#include "SurgeSynthesizer.h"
+
+namespace Surge
+{
+namespace Headless
+{
+
+SurgeSynthesizer* createSurge(int sr);
+
+typedef std::vector<float> audioData_t;
+
+void writeWavFile(const audioData_t& data, int nChannels, std::string fname);
+void writeGnuplotFile(const audioData_t& data, int nChannels, std::string fname);
+
+void renderDataTo(const audioData_t& data, int nChannels, std::ostream& str);
```

Comment:
`void writeToString(const void *data,  size_t size, int nChannels,  std::ostream& str);`

The typedef does make the code any cleaner. It only causes a distraction when you have to cross-reference it.  Here the raw data is probably the best option.

Reply:
Ooh `const void*` vs a type? I actually don’t agree with that choice! Why would you do that?

Reply:
It was a typo!

Reply:
Gotcha. Yeah so `const float* data, size_t size` vs `std:;vector<float>` - I prefer the vector if you don't particularly mind. Easier to do memory management and the like. Happy to change to that.

Reply:
I'm just a big fan of loose binding as far as it possible makes sense.  For example, you could use the same function to process only portion of the data without having to use iterators and such things. With either signature obviously the function does not do any memory management on the parameter.

Reply:
- Standard call would something like `writeToStream(result.data(), result.size(), 2, std::cout);`.
- Lets say you want to render `maxSamples` amount of samples is where you want to cut. Then you can just `writeToStream(result.data(), result.data() + maxSamples, 2, std::cout);`.

There is a lot robustness value to do this as you can probably see...

Reply:
OK you convinced me. But the accumulators still need a place to put the data; I'll keep a vector& there rather than a float** int* if that's OK with you.

The STL idiom of iterator from and to are a bit clunky. but we can insert the thousand year history of C and STL here.

Reply:
Actually you know what I'll just change it everywher